topics/basics/testing_plugins/integration_tests/integration_tests_api.md

```diff
@@ -0,0 +1,174 @@
+<!-- Copyright 2000-2025 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+
+# Integration Tests: API interaction
+
+<primary-label ref="2023.2"/>
+
+<link-summary>Walkthrough how to interact with API in integration tests.</link-summary>
+
+> This page is part of the [](integration_tests.md) tutorial.
+
+For introduction and setting up dependencies, refer to [](integration_tests_intro.md).
+
+## Java Management Extension
+
+The IDE and tests are running in different processes, which means some way to communicate between them is required.
+
+Driver framework uses Java Management Extensions (or JMX for short). JMX supports different connectors to manage the state of the JVM.
+Driver framework uses a standard Java Remote Method Invocation (RMI) protocol.
+This protocol allows accessing objects and invoking methods from tests in the JVM of the IDE.
+
+The architecture of RMI protocol is as follows:
+![](jmx_diagram.png)
```

Comment:
it would be better to use Mermaid diagram instead

Reply:
Or https://plantuml.com/sequence-diagram as Mermaid is sometimes limited with the possibilities.

Reply:
Oh, I didn't know that we support Mermaid. It's cool since this diagram is just an export of it into PNG.